tests/ui/coverage-attr/word-only.rs

```diff
@@ -0,0 +1,50 @@
+#![feature(coverage_attribute)]
+//@ edition: 2021
+
+// Demonstrates the diagnostics produced when using the syntax `#![coverage]`,
+// which should not be allowed.
```

Comment:
That test description seems wrong? Isn't it more about applying the attribute to the wrong item?

Reply:
This particular test is mainly about testing the error for `#[coverage]` in a variety of scenarios, most of which also happen to not permit coverage attributes of any kind.

It's motivated by the fact that the existing implementation was very inconsistent about how it handled this particular error case, giving different error messages in different places, and in some cases not issuing an error at all.

(Side note: I notice that I have an extra `!` that shouldn't be here.)